config/env_config.go

```diff
@@ -916,3 +923,11 @@ func (c EnvConfig) GetS3DisableExpressAuth() (value, ok bool) {
 
 	return *c.S3DisableExpressAuth, true
 }
+
+func (c EnvConfig) getEC2InstanceProfileName() (string, bool, error) {
+	if len(c.EC2InstanceProfileName) == 0 {
+		return "", false, nil
```

Comment:
It was discussed to err on this rather than leaving it empty. My main concern would be that someone would want to set this to something on their shell, like `export AWS_EC2_INSTANCE_PROFILE_NAME=$(some_process_that_returns_a_value)`, and that something returns an empty string, then they would make a request to the base URL rather than to the profile they were trying to use